palace.pyx

```diff
@@ -1208,6 +1214,118 @@ cdef class SourceGroup:
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl < (<AuxiliaryEffectSlot> other).impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl <= (<AuxiliaryEffectSlot> other).impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl == (<AuxiliaryEffectSlot> other).impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl != (<AuxiliaryEffectSlot> other).impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl > (<AuxiliaryEffectSlot> other).impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl >= (<AuxiliaryEffectSlot> other).impl
+
+    def __bool__(self) -> bool:
+        return <boolean> self.impl
+
+    def set_gain(self, value: float) -> None:
+        self.impl.set_gain(value)
+
+    def set_send_auto(self, value: bool) -> None:
+        self.impl.set_send_auto(value)
+
+    gain = property(fset=set_gain, doc=(
+        """Set gain for the effect slot.
+        """))
+    send_auto = property(fset=set_send_auto, doc=(
+        """If set to `True`, the reverb effect will automatically
+        apply adjustments to the source's send slot gains based
+        on the effect properties.
+
+        Has no effect when using non-reverb effects. Default is `True`.
+        """))
+
+    # TODO: apply effect
+
+    def destroy(self) -> None:
+        """Destroy the effect slot, returning it to the system.
+        If the effect slot is currently set on a source send,
+        it will be removed first.
+        """
+        return self.impl.destroy()
+
+    @property
+    def source_sends(self) -> Iterator[Tuple[Source, int]]:
+        """Retrieve each `Source` object and its pairing
+        send this effect slot is set on.
+        """
+        for source_send in self.impl.get_source_sends():
+            source = Source(None)
+            send = source_send.send
+            source.impl = source_send.source
+            yield source, send
+
+    @property
+    def use_count(self):
+        """Query the number of source sends the effect slot
+        is used by. This is equivalent to calling
+        `len(self.source_sends)`.
```

Comment:
Same here, i.e.  Number of...

Note that now `source_sends` is an iterator, it should be `len(tuple(self.source_sends))`.